src/libraries/System.Private.CoreLib/src/System/IO/FileStream.cs

```diff
@@ -107,27 +58,19 @@ public FileStream(IntPtr handle, FileAccess access, bool ownsHandle, int bufferS
                 throw;
             }
 
-            // Note: Cleaner to set the following fields in ValidateAndInitFromHandle,
-            // but we can't as they're readonly.
-            _access = access;
-            _useAsyncIO = isAsync;
-
-            // As the handle was passed in, we must set the handle field at the very end to
-            // avoid the finalizer closing the handle when we throw errors.
-            _fileHandle = safeHandle;
-        }
-
-        public FileStream(SafeFileHandle handle, FileAccess access)
-            : this(handle, access, DefaultBufferSize)
-        {
-        }
-
-        public FileStream(SafeFileHandle handle, FileAccess access, int bufferSize)
-            : this(handle, access, bufferSize, GetDefaultIsAsync(handle))
-        {
+            // it might seem to have no sense now, but we plan to introduce dedicated strategies for sync and async implementations
```

Comment:
Presumably at that point we're also going to factor in bufferSize, since for example bufferSize == 1 is the typical way to say "don't buffer".  At which point this switch as written is also insufficient.